### PR TITLE
Update jline to 3.23.0

### LINKED ISSUE #17915: Paste non-ASCII character into trino-cli causes Mojibake (garbled text)
Pasting a relatively long string with non-ASCII character into trino-cli causes part of the string becomes Mojibake (garbled text).

`jline` reads 64 bytes at a time. It will then decode that 64 bytes using `CharsetDecoder`.
When pasting a line contains more that 64bytes, it will repeat the above action until the end.

For non-ASCII character using UTF-8, a code point could take 1-4 bytes. For example, one Japanese character usually takes 3 bytes. If we paste 22 Japanese characters into trino-cli, that string is 22*3byte = 66 bytes. The last character will be broken.

```
First 64 bytes read: 21 Japanese characters + 1st byte of the last character
=> the 1st byte then being dropped silently

Second 64 bytes read: 2nd & 3rd byte of the last character
=> failed to decode and becomes Mojibake
```

How to reproduce:
Paste `一二三四五六七八九〇一二三四五六七八九〇一二三` into trino-cli (without using the slow pasting feature provided by terminal).
image

Environment:
* trino-cli-388 & trino-cli-419
* jline 3.21.0
* MacOS Monterey 12.6.1
* iTerm2 Build 3.4.19

Looks like https://github.com/jline/jline3/pull/782 fixed this issue unintentional(?).
Update `jline` to `3.22.0` fixed this issue on my machine.

### PR DESCRIPTION
**Pull request by oneonestar**
## Description
Udate `jline` to `3.23.0`

## Additional context and related issues
Fix #17915

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix garbled text caused by pasting string with non-ASCII character into trino-cli
```

### CODE REVIEW

**Review by ebyhr — DISMISSED** [review]

File: client/trino-cli/pom.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
         <project.build.targetJdk>8</project.build.targetJdk>
         <main-class>io.trino.cli.Trino</main-class>
-        <dep.jline.version>3.21.0</dep.jline.version>
+        <dep.jline.version>3.22.0</dep.jline.version>
```
[the remark this answers is not in the document]
Comment (oneonestar) [reply client/trino-cli/pom.xml:19]: https://github.com/jline/jline3/releases
Their Github homepage says `3.22.0` is the latest version but they released `3.23.0` in Maven.
I'll update to `3.23.0`.

**Comment by oneonestar** [comment]
```
$ java -jar /trino/client/trino-cli/target/trino-cli-420-SNAPSHOT-executable.jar > /tmp/stdout 2> /tmp/stderr

$ cat /tmp/stdout


$ cat /tmp/stderr
Jun 16, 2023 1:16:56 PM org.jline.utils.Log logr
WARNING: Unable to create a system terminal, creating a dumb terminal (enable debug logging for more information)
trino> %
```
Looks like for some reason the prompt went to stderr.

**Comment by oneonestar** [comment]
```
$ java -Djava.util.logging.config.file=/log.properties -jar /trino/client/trino-cli/target/trino-cli-420-SNAPSHOT-executable.jar > /tmp/stdout 2> /tmp/stderr

$ cat /tmp/stderr
Jun 16, 2023 1:31:57 PM org.jline.utils.Log logr
FINE: Unable to load JANSI support:
java.io.IOException: Unable to find terminal provider jansi
	at org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:78)
	at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:388)
	at org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:314)
	at io.trino.cli.TerminalUtils.createTerminal(TerminalUtils.java:44)
	at io.trino.cli.TerminalUtils.<clinit>(TerminalUtils.java:28)
	at io.trino.cli.Console.run(Console.java:128)
	at io.trino.cli.Console.call(Console.java:100)
	at io.trino.cli.Console.call(Console.java:73)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at io.trino.cli.Trino.main(Trino.java:42)

Jun 16, 2023 1:31:57 PM org.jline.utils.Log logr
WARNING: input is tty: {}true
Jun 16, 2023 1:31:57 PM org.jline.utils.Log logr
WARNING: output is tty: {}false
Jun 16, 2023 1:31:57 PM org.jline.utils.Log logr
WARNING: error is tty: {}false
Jun 16, 2023 1:31:57 PM org.jline.utils.Log logr
WARNING: Creating a dumb terminal
java.lang.IllegalStateException: Unable to create a terminal
	at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:375)
	at org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:314)
	at io.trino.cli.TerminalUtils.createTerminal(TerminalUtils.java:44)
	at io.trino.cli.TerminalUtils.<clinit>(TerminalUtils.java:28)
	at io.trino.cli.Console.run(Console.java:128)
	at io.trino.cli.Console.call(Console.java:100)
	at io.trino.cli.Console.call(Console.java:73)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at io.trino.cli.Trino.main(Trino.java:42)
	Suppressed: java.io.IOException: Unable to find terminal provider jansi
		at org.jline.terminal.spi.TerminalProvider.load(TerminalProvider.java:78)
		at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:388)
		... 15 more

Jun 16, 2023 1:31:57 PM org.jline.utils.Log logr
FINE: Using terminal DumbTerminal
trino> Jun 16, 2023 1:31:58 PM org.jline.utils.Log logr
FINE: NonBlockingInputStream start
Jun 16, 2023 1:32:21 PM org.jline.utils.Log logr
FINE: NonBlockingInputStream shutdown
```
Looks like we need to config `TerminalProvider`.

**Comment by wendigo** [comment]
JLine was just upgraded to 3.24.0

**Pull request closed without merging**